akka23-conductr-client-lib/src/main/scala/com/typesafe/conductr/clientlib/akka/ControlClient.scala

```diff
@@ -0,0 +1,368 @@
+ */
+object ControlClient {
+
+  /** SCALA API **/
+  def apply(conductrAddress: URL): ControlClient =
+    new ControlClient(new ConnectionHandler, conductrAddress, ApiVersion.V2)
+
+  /** JAVA API **/
+  def getInstance(conductrAddress: URL): ControlClient =
+    ControlClient(conductrAddress)
+}
+
+/**
+ * Akka 2.3 flavor of the [[com.typesafe.conductr.clientlib.scala.AbstractControlClient]].
+ * Uses Akka HTTP and Akka streams to interact with ConductR.
+ * @param handler The connection handler contains the logic how to interact with the ConductR HTTP server
+ * @param conductrAddress contains the protocol://host:port of the ConductR control server
+ * @param apiVersion The version of ConductR's REST API
+ */
+private[akka] class ControlClient(handler: ConnectionHandler, conductrAddress: URL, apiVersion: ApiVersion.Value)
+    extends AbstractControlClient(conductrAddress) {
+
+  import EventStreamUnmarshalling._
+  import JsonMarshalling._
+
+  override protected type CC = ConnectionContext
+
+  /**
+   * @see [[AbstractControlClient.getBundlesInfo()]]
+   * @param cc implicit connection context
+   * @return the bundles
+   */
+  override def getBundlesInfo()(implicit cc: ConnectionContext): Future[Seq[Bundle]] =
+    handler.withConnectedRequest(Payload.bundlesInfo)(handleAsSeq[Bundle])
+
+  /**
+   * @see [[AbstractControlClient.loadBundle()]]
+   * @param bundle The file that is the bundle.
+   *               The filename is important with its hex digest string and is required to be consistent
+   *               with the SHA-256 hash of the bundle’s contents.
+   *               Any inconsistency between the hashes will result in the load being rejected.
+   * @param config Optional: Similar in form to the bundle, only that is the file that describes the configuration.
+   *               Again any inconsistency between the hex digest string in the filename, and the SHA-256 digest
+   *               of the actual contents will result in the load being rejected.
+   * @param cc implicit connection context
+   * @return The result as a Future[BundleRequestResult]. BundleRequestResult is a sealed trait and can be either:
+   *         - BundleRequestSuccess if the loading request has been succeeded. This object contains the request and bundle id
+   *         - BundleRequestFailure if the loading request has been failed. This object contains the HTTP status code and error message.
+   */
+  override def loadBundle(bundle: URI, config: Option[URI])(implicit cc: ConnectionContext): Future[BundleRequestResult] = {
+    def createRequestBody: Future[RequestEntity] = {
+      import cc.actorMaterializer.executionContext
+
+      // TODO: Use Akka stream to extract and read the zip file.
+      //       In the current Akka streams version 2.0.1 there is no utility to extract a zip file and iterate over
+      //       the files inside of the zip file. Therefore a blocking solution inspired by sbt.IO is used.
```

Comment:
ConductR offers a better solution within `BundleWriter`'s `extractZip` function. Also, where is is stated that streams will be providing such a function?